lib/ssh/src/ssh_acceptor_subsup.erl

```diff
@@ -0,0 +1,142 @@
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% %CopyrightEnd%
+%%
+
+%%
+%%----------------------------------------------------------------------
+%% Purpose: The acceptor supervisor for ssh servers hangs under 
+%%          ssh_system_sup.
+%%----------------------------------------------------------------------
+
+-module(ssh_acceptor_subsup).
+-moduledoc false.
+-behaviour(supervisor).
+
+-include("ssh.hrl").
+
+-export([start_link/3,
+         listen/2
+        ]).
+
+%% Supervisor callback
+-export([init/1]).
+
+%%%=========================================================================
+%%%  API
+%%%=========================================================================
+start_link(SystemSup, Address, Options) ->
+    case supervisor:start_link(?MODULE, [SystemSup, Address, Options]) of
+        {error, {shutdown, {failed_to_start_child, _, Error}}} ->
+            {error,Error};
+        Other ->
+            Other
+    end.
+
+%%%=========================================================================
+%%%  Supervisor callback
+%%%=========================================================================
+init([SystemSup, Address=#address{port=Port}, Options]) ->
+    ssh_lib:set_label(server, acceptor_subsup),
+    %% Initial start of ssh_acceptor_sup for this port
+    Self = self(),
+    NumAcceptors = case ?GET_OPT(parallel_login, Options) of
+                       true -> 1;
+                       false -> 1;
+                       N -> N
+                   end,
+    {Sock, Opts} = case ?GET_INTERNAL_OPT(lsocket, Options, undefined) of
+        {LSock, SockOwner} ->
+            %% A listening socket (or fd option) was provided in the ssh:daemon call
+            case inet:sockname(LSock) of
+                {ok, {_, Port}} ->
+                    %% A usable, open LSock
+                    spawn_link(fun() ->
+                              SockOwner ! {request_control, LSock, Self, self()},
+                              receive {its_yours, LSock} -> ok end,
+                              [supervisor:start_child(Self, []) || _ <- lists:seq(1, NumAcceptors)],
+                              unlink(Self)
+                          end),
+                    {LSock, Options};
+
+                {error, _Error} ->
+                    %% Not open, a restart
+                    %% Allow gen_tcp:listen to fail 4 times if eaddrinuse (It is a bug fix):
+                    case try_listen(Port, Options, 4) of
+                        {ok, NewLSock} ->
+                            spawn_link(fun() ->
```

Comment:
Why do we do spawn_link and unlink? We could do just spawn if we do not want the link?

Reply:
Oh, that's a relic :sweat_smile: Initially, I had more stuff going on in this function, and the `spawn_link` was there to stop the spawned process in case the supervisor exits while it was doing that. The `unlink` was there to prevent the supervisor receiving and complaining about an unexpected `'EXIT'` message when the spawned process exits normally.

tl;dr: I'll change it as you suggest, also on lines 96 and 98.

Reply:
done :+1: